specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json

```diff
@@ -9444,6 +9444,18 @@
         "primary": {
           "type": "boolean",
           "description": "Specifies the primary network interface in case the virtual machine has more than 1 network interface."
+        },
+        "deleteOption": {
+          "type": "string",
+          "description": "Specify what happens to the network interface when the VM is deleted",
+          "enum": [
+            "Delete",
+            "Detach"
+          ],
+          "x-ms-enum": {
+            "name": "DeleteOptions",
+            "modelAsString": false
```

Comment:
We can sync with Jun about best practice, but my understanding is that modeling enums as strings is current best pratice to allow for future additios to the enum.

If not modeled as a string new enums reult in deserialization errors in old clients